src/core/thread/mle_router.cpp

```diff
@@ -2515,10 +2515,6 @@ void MleRouter::HandleChildUpdateRequest(const Message &         aMessage,
         {
             child->SetCslChannel(static_cast<uint8_t>(cslChannel.GetChannel()));
         }
```

Comment:
If `kCslChannel` Tlv. could not be found, should the parent clears the child's csl channel?

Reply:
I think it shouldn't.

After reading this again:
> If configured, CSL data transmission and reception using the additional channel operates as described in Section 6.12.2.6 in [IEEE802154-2015]. The CSL Channel for a Thread Interface operating as a CSL Receiver or as a CSL Transmitter is either:
> • The value of phyCurrentChannel when macCSLChannelMask has value 0.
> • The channel value selected by macCSLChannelMask when macCSLChannelMask is different from 0.

I think it's ok to set for SSED to set CSL Channel to 0 again if it wants so. So when there's no CSL Channel TLV, we don't need to clear child's CSL channel.

Reply:
See CSL Channel TLV definition in 4.4.21
> IEEE 802.15.4 CslChannelPage: An unsigned 8-bit value that specifies the channel page.
> IEEE 802.15.4 CslChannel: An unsigned 16-bit value that identifies the channel within the channel page.

I am afraid channel 0 is is not a valid channel within the channel page 0, which have 16 channels [11, 26] in the 2450 MHz band

Reply:
@librasungirl 

After reading the spec again, I have some interesting findings and questions now. Correct me if I'm wrong.

First I must say, currently my description of **'CSL channel has been specified or not'** is inaccurate. The spec actually means:
1. When *macCSLChannelMask* has value `0`, CSL tx uses `phyCurrentChannel`.
2. When *macCSLChannelMask* isn't `0`, CSL tx uses `phyCurrentChannel`.

**I would call these two conditions case 1 and case 2 below.**

*macCSLChannelMask* is a PIB attribute. So if we set CSL Channel first(move to case 2) and then want to get back to case 1, we just need to change *macCSLChannelMask* back to 0. I believe that `The value of phyCurrentChannel when macCSLChannelMask has value 0.` indicates *macCSLChannelMask* could equal to 0.

The spec also mentions that(**3.2.6.3.2**):

> The macCslChannelMask is relayed by a CSL Receiver SSED Child to a CSL Transmitter Parent Router via the CSL Channel TLV sent in the Child Update Request as described in Section 4.6

However, what you mentioned(CSL Channel TLV in 4.4.31) is right. If we set `CslChannelPage` to `0`, then we cannot set `CslChannel` to `0` in the TLV.

So here is my question:
1. I found it's mentioned in **8.10.1.1.1 Supported Channel Pages**, currently only `0` is a valid value, `1-255` are reserved. Can we use `1-255` for our implementation usage?
2. If by no means we can relay `0` in CSL Channel TLV, then I think we cannot use a 'Thread' way to move from case 2 to case 1. We have to set *macCSLChannelMask* on both sides if we want to do so. Actually I think it's fine even if we don't support to move from case 2 to case 1. Just ask for your opinion.

For now, I would prefer implement it all conforming to the spec. That is, we cannot move from case 2 to case 1. Thoughts?

Reply:
channel page 1-255 for 2.4Ghz conflicts IEEE802.15.4 spec.

Not sure how to exactly interprete the `relay` here,  macCSLChannelMask is not supported in your CSL implementation, in fact, even supported, the mask should also only represent channels [11, 26], and choose one within these channels. I think so far we may simply focus on the CSL Channel TLV and may assume user can specify any channel valid for channel page 0?

I think it should be fine to just not append CSL channel Tlv when unspecified, and clear the SSED's csl channel when the parent receives CUR without CSL channel TLV.

Thoughts?

Reply:
> I think it should be fine to just not append CSL channel Tlv when unspecified, and clear the SSED's csl channel when the parent receives CUR without CSL channel TLV.

SGTM.